sync/src/block/extension.rs

```diff
@@ -690,8 +708,48 @@ impl Extension {
     }
 }
 
-impl BlockSyncInfo for Extension {
-    fn get_peers(&self) -> Vec<NodeId> {
-        self.header_downloaders.read().keys().cloned().collect()
+pub struct BlockSyncSender(EventSender<Event>);
+
+impl From<EventSender<Event>> for BlockSyncSender {
+    fn from(sender: EventSender<Event>) -> Self {
+        BlockSyncSender(sender)
+    }
+}
+
+impl ChainNotify for BlockSyncSender {
+    fn new_headers(
```

Comment:
@Kais-DkM Should ChainNotify be synchronous?
Is it okay not to wait for the result of the event?

Reply:
`new_headers` and `new_blocks` creates all the lock they need.
It should be safe to call them asynchronously.